modules/p2/ding_list/ding_list.module

```diff
@@ -0,0 +1,987 @@
+        $vars['row_attributes'][]['data-item-id'] = $item->id;
+      }
+
+      $path = drupal_get_path('module', 'ding_list');
+      drupal_add_tabledrag($table_id, 'order', 'sibling', 'views-field-delta');
+      drupal_add_js($path . '/js/ding_list_tabledrag.js');
+      drupal_add_css($path . '/css/ding-list-tabledrag.css');
+      drupal_add_css($path . '/css/ding-list-view.css');
+    }
+    // Hide dragging fields if dragging is not enabled.
+    else {
+      if (isset($vars['field_classes']['id'])) {
+        foreach ($vars['field_classes']['id'] as &$classes) {
+          $classes .= ' element-hidden';
+        }
+      }
+      
+      if (isset($vars['field_classes']['delta'])) {
+        foreach ($vars['field_classes']['delta'] as &$classes) {
+          $classes .= ' element-hidden';
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Change the html header for lists, to ensure og:url and og:description.
+ */
+function ding_list_html_head_alter(&$head_elements) {
+  // If the current page is a list.
+  if (arg(0) === 'list' && is_numeric(arg(1)) && !arg(2)) {
+    $list = ding_list_get_list(arg(1));
+    $head_elements['ogurl'] = array(
+      '#type' => 'html_tag',
+      '#tag' => 'meta',
+      '#attributes' => array(
+        'property' => 'og:url',
+        'content' => url('list/' . arg(1), array('absolute' => TRUE)),
+      ),
+    );
+    $head_elements['ogdescription'] = array(
+      '#type' => 'html_tag',
+      '#tag' => 'meta',
+      '#attributes' => array(
+        'property' => 'og:description',
+        'content' => ding_base_get_value('ding_type', $list, 'field_notes'),
+      ),
+    );
+  }
+}
+
+/**
+ * Implements hook_entity_load().
+ *
+ * When a list is loaded convert the cql of ting_objects to the new cql.
+ */
+function ding_list_entity_load($entities, $type) {
+  $saved = &drupal_static(__FUNCTION__, array());
+  if ($type === 'ding_type') {
+    foreach ($entities as & $entity) {
+      if ($entity->type === 'ding_list_element' && !isset($saved[$entity->id])) {
+        if (ding_base_get_value('ding_type', $entity, 'field_element_type') === 'ting_object') {
+          $value = ding_base_get_value('ding_type', $entity, 'field_value');
+          if (empty($value)) {
+            $value = ding_base_get_value('ding_type', $entity, 'field_ting_object');
+          }
+
+          $p2value = ding_list_convert_cql($value);
+          if ($value !== $p2value) {
+            $saved[$entity->id] = TRUE;
+            $values = array(
+              'value' => $p2value,
+            );
+            ding_list_edit_ding_list_element($entity, 'ting_object', $values);
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Convert old cql.
+ *
+ * We need to convert old cql to new cql. Old cql will occur if ding_list has
+ * been used on an environment using the old cql before upgrading.
+ *
+ * This function is needed, because the cql converter included in ding core is
+ * incomplete.
```

Comment:
I still think it is relevant to determine why the current CQL converter is incomplete.

Even if we keep this function around for now there is no chance of it ever going away if noone is aware what it takes to migrate to the core implementation.